### PR TITLE
Fix #116 param bugs (based on @monkoose patch)

### LINKED ISSUE #116: Get rid of param from completions
Some enhancement suggestion.
Before jedi version 0.10 in completions we had something like this
`math.pow(x, y)`
After jedi version 0.10 we have this `param` word before every parameter
`math.pow(param x, param y)`
So completions get too long and actually distracting. Would be much better to ignore them.

### PR DESCRIPTION
**Pull request by Shougo**
It fixes jedi param bugs in #116.

@monkoose Please check it.

### CODE REVIEW

File: rplugin/python3/deoplete/sources/deoplete_jedi/server.py [hunk 1]
```diff
@@ -395,7 +395,10 @@ def completion_dict(self, name, type_, comp):
                         continue
                     if '\\n' in desc:
                         desc = desc.replace('\\n', '\\x0A')
-                    params.append(desc)
+                    if desc.startswith('param'):
+                        desc = desc[5:].strip()
```
Comment (blueyed) [inline rplugin/python3/deoplete/sources/deoplete_jedi/server.py]: Shouldn't that check for `'param '` (with space at the end), and then remove 6 chars?

**Comment by blueyed** [comment]
Just for reference: it seems to come from here: https://github.com/davidhalter/jedi/blob/5427b02712828b2875d35b5ee1c8b5e58f820537/jedi/api/classes.py#L542-L543 ?!

@davidhalter
Is this the correct way to approach this?  (getting rid of the `param` prefix for params)

**Comment by Shougo** [comment]
Fixed.

**Comment by monkoose** [comment]
> Just for reference: it seems to come from here: https://github.com/davidhalter/jedi/blob/5427b02712828b2875d35b5ee1c8b5e58f820537/jedi/api/classes.py#L542-L543 ?!

So as i said - its a feature. But does @davidhalter know that it is looking ugly in completion engines?

@Shougo as work patch - works perfectly for me. And the part `if desc:` - removes blank-wrong parameters just as needed.
But i don't sure you need to check explicitly `if desc.startswith('param ') or desc == 'param':` In my opinion `if desc.startswith('param')` does the same thing.

**Comment by Shougo** [comment]
> But i don't sure that you need to check explicitly `if desc.startswith('param ') or desc == 'param'`:

This is hack.
So it should be checked explicitly.
It may be fixed in the later version.

**Comment by monkoose** [comment]
Ok. I agree.

**Comment by davidhalter** [comment]
> @davidhalter
Is this the correct way to approach this? (getting rid of the param prefix for params)

The problem is actually that there is no API for this. What we've been using was the description. The description IMO is just there to give you a better impression with what you're dealing with. If you just want the name of a parameter, there's an API for that. But there's definitely no API for stuff like **kwargs.

So in conclusion if you want that, try to propose an API for it :)

**Comment by Shougo** [comment]
Merged
